pkg/security/probe/probe_auditing_windows.go

```diff
@@ -0,0 +1,151 @@
+<data name="ObjectType" inType="win:UnicodeString"/>
+<data name="ObjectName" inType="win:UnicodeString"/>
+<data name="HandleId" inType="win:Pointer"/>
+<data name="OldSd" inType="win:UnicodeString"/>
+<data name="NewSd" inType="win:UnicodeString"/>
+<data name="ProcessId" inType="win:Pointer"/>
+<data name="ProcessName" inType="win:UnicodeString"/>
+</template>
+*/
+
+// we're going to try for a slightly more useful name
+//
+//revive:disable:var-naming
+type objectPermsChange struct {
+	etw.DDEventHeader
+	subjectUserSid    string
+	subjectUserName   string
+	subjectDomainName string
+	subjectLogonId    string
+	objectServer      string
+	objectType        string
+	objectName        string
+	handleId          fileObjectPointer
+	oldSd             string
+	newSd             string
+	processId         fileObjectPointer
+	processName       string
+}
+
+func (wp *WindowsProbe) parseObjectPermsChange(e *etw.DDEventRecord) (*objectPermsChange, error) {
+
+	pc := &objectPermsChange{
+		DDEventHeader: e.EventHeader,
+	}
+	data := etwimpl.GetUserData(e)
+
+	reader := stringparser{nextRead: 0}
+	pc.subjectUserSid = reader.GetSIDString(data)
+	pc.subjectUserName = reader.GetNextString(data)
+	pc.subjectDomainName = reader.GetNextString(data)
+	pc.subjectLogonId = strconv.FormatUint(reader.GetUint64(data), 16)
+	pc.objectServer = reader.GetNextString(data)
+	pc.objectType = reader.GetNextString(data)
+	pc.objectName = reader.GetNextString(data)
+
+	pc.handleId = fileObjectPointer(reader.GetUint64(data))
+
+	pc.oldSd = reader.GetNextString(data)
+	pc.newSd = reader.GetNextString(data)
+
+	pc.processId = fileObjectPointer(reader.GetUint64(data))
+
+	pc.processName = reader.GetNextString(data)
+
+	// translate the registry path, if it's a registry path, into the more canonical form
+	pc.objectName = translateRegistryBasePath(pc.objectName)
+	return pc, nil
+}
+
+func (pc *objectPermsChange) String() string {
+	var output strings.Builder
+	output.WriteString(" ObjectPermsChange name: " + pc.objectName + "\n")
+	output.WriteString("                   oldsd: " + pc.oldSd + "\n")
+	output.WriteString("                   newsd: " + pc.newSd + "\n")
+
+	return output.String()
+}
+
+type stringparser struct {
+	nextRead int
+}
+
+func (sp *stringparser) GetNextString(data etw.UserData) string {
+	s, no, _, _ := data.ParseUnicodeString(sp.nextRead)
+
+	if no == -1 {
+		sp.nextRead += 2
+	} else {
+		sp.nextRead = no
+	}
+	return s
+}
+
+func (sp *stringparser) GetSIDString(data etw.UserData) string {
+	l := data.Length()
+	b := data.Bytes(sp.nextRead, l-sp.nextRead)
+	sid := (*windows.SID)(unsafe.Pointer(&b[0]))
+	sidlen := windows.GetLengthSid(sid)
+	sp.nextRead += int(sidlen)
+
+	var winstring *uint16
+	err := windows.ConvertSidToStringSid(sid, &winstring)
+	if err != nil {
+		return ""
+	}
+	defer windows.LocalFree(windows.Handle(unsafe.Pointer(winstring)))
+
+	return windows.UTF16PtrToString(winstring)
+
+}
```

Comment:
I wonder if we should maintain a quicker cache since there are not that many SIDs around